### PR TITLE
Fix UniConn bufferSize bug and implementment of close read unit test

### PR DESCRIPTION
**Pull request by billfort**
1. Fix bug of UniConn bufferSize 
2. Rest readCtx and writeCtx after Deadline expired
3. Implement of CloseRead and CloseWrite of NetConn
4. Arrangement of unit public functions

Signed-off-by: billfort <fxbao@hotmail.com>

### CODE REVIEW

File: uniconn.go [hunk 1]
```diff
@@ -93,13 +98,26 @@ func (uc *UniConn) Write(b []byte) (n int, err error) {
 		return 0, ErrZeroLengh
 	}
 
+	var timeoutCtx context.Context
+	var timeoutCancel context.CancelFunc
+	if uc.writeTimeout > 0 {
+		timeoutCtx, timeoutCancel = context.WithTimeout(context.Background(), uc.writeTimeout)
```
Comment (yilunzhang) [inline uniconn.go]: We can change background to uc.writeCtx so we can have one less case in select

File: uniconn.go [hunk 2]
```diff
@@ -190,14 +205,24 @@ func (uc *UniConn) Read(b []byte) (n int, err error) {
 		}
 	}
 
+	var timeoutCtx context.Context
+	var timeoutCancel context.CancelFunc
+	if uc.readTimeout > 0 {
+		timeoutCtx, timeoutCancel = context.WithTimeout(context.Background(), uc.readTimeout)
```
Comment (yilunzhang) [inline uniconn.go]: We can change background to uc.readCtx so we can have one less case in select

File: uniconn.go (hunk 1 above)
Comment (billfort) [reply uniconn.go]: Yes, modified

File: uniconn.go (hunk 2 above)
Comment (billfort) [reply uniconn.go]: Yes, modified

**Review by yilunzhang — APPROVED** [review]

**Pull request merged**